### ISSUE TITLE
[ISSUE] databricks_sql_endpoint does not contain spot_instance_policy

### ISSUE THREAD

**Issue by KoenR3** [databrickslabs/terraform-provider-databricks#665]
Hi there,

### Affected Resource(s)
Please list the resources as a list, for example:
- databricks_sql_endpoint

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "databricks_sql_endpoint" "this" {
  name = "Dashboarding Endpoint"
  cluster_size = "2X-Small"
  max_num_clusters = 1
  auto_stop_mins = 60
  instance_profile_arn = var.dashboard_endpoint_instance_arn
  spot_instance_policy = "COST_OPTIMIZED"
  enable_photon = true
}
```

### Debug Output
```
2021-05-26T11:51:23.843+0200 [DEBUG] provider.terraform-provider-databricks_v0.3.4: Version 0.3.4
2021-05-26T11:51:23.843+0200 [DEBUG] provider.terraform-provider-databricks_v0.3.4: 
2021-05-26T11:51:23.843+0200 [DEBUG] provider.terraform-provider-databricks_v0.3.4: https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs
2021-05-26T11:51:23.843+0200 [DEBUG] provider.terraform-provider-databricks_v0.3.4: 
2021-05-26T11:51:23.844+0200 [INFO]  provider.terraform-provider-databricks_v0.3.4: configuring server automatic mTLS: timestamp=2021-05-26T11:51:23.844+0200
2021-05-26T11:51:23.897+0200 [DEBUG] provider.terraform-provider-databricks_v0.3.4: plugin address: address=/var/folders/2h/szc16rjx6lb96x0_2kww5k1h0000gn/T/plugin054663595 network=unix timestamp=2021-05-26T11:51:23.897+0200
2021-05-26T11:51:23.897+0200 [DEBUG] provider: using plugin: version=5
╷
│ Error: Unsupported argument
│ 
│   on ../../modules/sql_analytics/endpoints.tf line 7, in resource "databricks_sql_endpoint" "this":
│    7:   spot_instance_policy = "COST_OPTIMIZED"
│ 
│ An argument named "spot_instance_policy" is not expected here.
╵
2021-05-26T11:51:23.973+0200 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-05-26T11:51:23.977+0200 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/databrickslabs/databricks/0.3.4/darwin_amd64/terraform-provider-databricks_v0.3.4 pid=97212
2021-05-26T11:51:23.977+0200 [DEBUG] provider: plugin exited

```

### Expected Behavior
According to the documentation, the endpoint has a argument `spot_instance_policy` which is an option on the endpoint.  However, it is not in the code.  Not sure if I need to log this as a documentation issue or a provider issue.  I decided for provider as it is expected that you can set this option.

### Steps to Reproduce
Provision a SQL analytics endpoint with the argument `spot_instance_policy`

**Comment by alexott** [comment]
it will be fixed with #666

**Comment by alexott** [comment]
fixed
